extensions/vault/runtime/src/main/java/io/quarkus/vault/internal/client/VaultClient.java

```diff
@@ -0,0 +1,44 @@
+package io.quarkus.vault.internal.client;
+
+import io.quarkus.vault.internal.client.dto.VaultDbCreds;
+import io.quarkus.vault.internal.client.dto.VaultInit;
+import io.quarkus.vault.internal.client.dto.VaultInitBody;
+import io.quarkus.vault.internal.client.dto.VaultKubernetesAuth;
+import io.quarkus.vault.internal.client.dto.VaultKubernetesAuthBody;
+import io.quarkus.vault.internal.client.dto.VaultKvSecretV1;
+import io.quarkus.vault.internal.client.dto.VaultKvSecretV2;
+import io.quarkus.vault.internal.client.dto.VaultLeasesBody;
+import io.quarkus.vault.internal.client.dto.VaultLeasesLookup;
+import io.quarkus.vault.internal.client.dto.VaultLookupSelf;
+import io.quarkus.vault.internal.client.dto.VaultRenewLease;
+import io.quarkus.vault.internal.client.dto.VaultRenewSelf;
+import io.quarkus.vault.internal.client.dto.VaultRenewSelfBody;
+import io.quarkus.vault.internal.client.dto.VaultUserPassAuth;
+import io.quarkus.vault.internal.client.dto.VaultUserPassAuthBody;
+
+public interface VaultClient {
+
+    String X_VAULT_TOKEN = "X-Vault-Token";
+    String API_VERSION = "v1";
+
+    VaultUserPassAuth loginUserPass(String user, VaultUserPassAuthBody body);
+
+    VaultKubernetesAuth loginK8s(VaultKubernetesAuthBody body);
+
+    VaultRenewSelf renewSelf(String token, VaultRenewSelfBody body);
+
+    VaultLookupSelf lookupSelf(String token);
+
+    VaultLeasesLookup lookupLease(String token, VaultLeasesBody body);
+
+    VaultRenewLease renewLease(String token, VaultLeasesBody body);
+
+    VaultKvSecretV1 getSecretV1(String token, String secretEnginePath, String path);
+
+    VaultKvSecretV2 getSecretV2(String token, String secretEnginePath, String path);
+
+    VaultDbCreds generateDbCredentials(String token, String dbrole);
+
+    VaultInit init(VaultInitBody body);
```

Comment:
@vsevel, thanks for removing a couple of other methods unused in the main source, but I do think this `init` method should be gone from the public interface, with 2 less dto classes, it is worth it. If you prefer to have the test code emulating the real sequence, then maybe have a `TestVaultClient` extending that `VaultClient`, with the `init` and two `dto` objects being the test source for now. 
I doubt a lot, that even if `VaultClient` becomes directly available to the users, we'd ask them to start initializing the vault, they will be working against the vaults prepared for them.

Reply:
> but I do think this init method should be gone from the public interface

well it is not a public interface... I think it is overkill to chase those last methods since it is internal stuff and those methods are part of the official vault api, as long as we provide tests to make sure the signature is valid.
I dropped it anyway, and added a TestVaultClient as you suggested. I took this opportunity to re-add the getHealth() and getSealStatus() on the test side. I could not use it in VaultITCase like before, because the vault client we get in VaultITCase is the real one, not TestVaultClient. so I moved those methods to VaultTestExtension.

> I doubt a lot, that even if VaultClient becomes directly available to the users, we'd ask them to start initializing the vault, they will be working against the vaults prepared for them.

I agree. except if vault client is used like kubernetes client, for instance in a cli. then we will definitely all the admin apis.